Source/Scene/Expression.js

```diff
@@ -0,0 +1,856 @@
+        defineProperties,
+        DeveloperError,
+        jsep,
+        ExpressionNodeType) {
+    "use strict";
+
+    var unaryOperators = ['!', '-', '+'];
+    var binaryOperators = ['+', '-', '*', '/', '%', '===', '!==', '>', '>=', '<', '<=', '&&', '||', '!~', '=~'];
+
+    var variableRegex = /\${(.*?)}/g;
+    var backslashRegex = /\\/g;
+    var backslashReplacement = '@#%';
+    var replacementRegex = /@#%/g;
+
+    var scratchColor = new Color();
+
+    /**
+     * DOC_TBA
+     */
+    function Expression(expression) {
+        //>>includeStart('debug', pragmas.debug);
+        if (typeof(expression) !== 'string') {
+            throw new DeveloperError('expression must be a string.');
+        }
+        //>>includeEnd('debug');
+
+        expression = replaceVariables(removeBackslashes(expression));
+
+        // customize jsep operators
+        jsep.addBinaryOp("=~", 0);
+        jsep.addBinaryOp("!~", 0);
+
+        var ast;
+        try {
+            ast = jsep(expression);
+        } catch (e) {
+            //>>includeStart('debug', pragmas.debug);
+            throw new DeveloperError(e);
+            //>>includeEnd('debug');
+        }
+
+        this._runtimeAst = createRuntimeAst(this, ast);
+    }
+
+    defineProperties(Expression.prototype, {
+
+        // TODO: Expose AST?
+
+    });
+
+    /**
+     * Evaluates the result of an expression, using the values defined by a feature. If the result is of type
+     * <code>Boolean</code>, <code>Number</code>, or <code>String</code>, the corresponding primitive type will be
+     * returned. If the result is a <code>RegExp</code>, a Javascript <code>RegExp</code> object will be returned. If
+     * the result is a <code>Color</code>, a Cesium <code>Color</code> object will be returned.
+     *
+     * @param {Cesium3DTileFeature} feature The feature of which values are used to evaluate the expression.
+     * @returns {Boolean|Number|String|Color|RegExp} The result of the expression.
+     */
+    Expression.prototype.evaluate = function(feature) {
+        return this._runtimeAst.evaluate(feature);
+    };
+
+    /**
+     * Evaluates the result of a Color expression, using the values defined by a feature.
+     *
+     * @param {Cesium3DTileFeature} feature The feature of which values are used to evaluate the expression.
+     * @param {Color} [result] The object in which to store the result
+     * @returns {Color} The modified result parameter or a new Color instance if one was not provided.
+     */
+    Expression.prototype.evaluateColor = function(feature, result) {
+        return this._runtimeAst.evaluate(feature, result);
+    };
+
+    function Node(type, value, left, right, test) {
+        this._type = type;
+        this._value = value;
+        this._left = left;
+        this._right = right;
+        this._test = test;
+        this.evaluate = undefined;
+
+        setEvaluateFunction(this);
+    }
+
+    // TODO: Make fix to jsep to allow backslashes
```

Comment:
@ggetz if the TODOs in this file are already covered in the roadmap, remove them.  Otherwise, add them to the roadmap and remove them.